extensions/amp-ad-exit/0.1/amp-ad-exit.js

```diff
@@ -190,6 +257,36 @@ export class AmpAdExit extends AMP.BaseElement {
       this.user().error(TAG, 'Invalid JSON config', e);
       throw e;
     }
+
+    this.unlisten_ = listen(this.getAmpDoc().win, 'message', event => {
+      const responseMessage = deserializeMessage(getData(event));
+
+      let ampAdResourceId;
+      try {
+        ampAdResourceId = this.element.ownerDocument.defaultView
+          .frameElement.parentElement.getResourceId();
+      } catch (e) {
+        this.user().error(TAG,
```

Comment:
I've never seen `this.user()` before. I should probably ask Hongfei or somebody from AMP Core what it's for...

Reply:
I think this was just a careless mistake. Removing.

Reply:
No, it's used elsewhere in this file.

Reply:
Hmm...I do see lots of occurrences: https://github.com/ampproject/amphtml/search?utf8=%E2%9C%93&q=%22this.user%28%29.error%22&type=

Reply:
We migrated to `this.user().error(` which equals to `user(this.element).error(`. With the change we can tell if the user error is from the parent AMP doc or from the friendly iframe.